src/rust/ensogl/lib/components/src/scrollbar.rs

```diff
@@ -0,0 +1,283 @@
+                size / max);
+            // Minimum thumb size in normalized units
+            min_visual_size     <- inner_length.map(|&length| MIN_THUMB_SIZE / length);
+            // The size at which we render the thumb on screen, in normalized units. Can differ from
+            // the actual thumb size if the thumb is smaller than the min.
+            visual_size         <- all_with(&normalized_size,&min_visual_size,|&size,&min|
+                size.max(min).min(1.0));
+            // The position at which we render the thumb on screen, in normalized units.
+            visual_start        <- all_with(&normalized_position,&visual_size,|&pos,&size|
+                pos * (1.0 - size));
+            visual_bounds       <- all_with(&visual_start,&visual_size,|&start,&size|
+                Bounds::new(start,start+size));
+            visual_center       <- visual_bounds.map(|bounds| bounds.center());
+            thumb_center_px     <- all_with(&visual_center,&inner_length, |normalized,length|
+                (normalized - 0.5) * length);
+
+            update_slider <- all(&visual_bounds,&resize);
+            eval update_slider(((value,size)) model.set_background_range(*value,*size));
+
+
+            // === Clicking ===
+
+            frp.scroll_by <+ base_frp.background_click.map3(&thumb_center_px,&frp.set_thumb_size,
+                |click_position,thumb_center,thumb_size| {
+                    let direction = if click_position.x > *thumb_center { 1.0 } else { -1.0 };
+                    direction * thumb_size * CLICK_JUMP_PERCENTAGE
+                });
+
+
+            // === Dragging ===
+
+            drag_started <- base_frp.is_dragging_track.on_change().on_true().constant(());
+            x            <- all4(&mouse_position,&inner_length,&frp.set_max,&frp.thumb_position);
+            x            <- x.sample(&drag_started);
+            drag_offset  <- x.map(|(mouse_px,length_px,max,thumb_pos)| {
+                    let thumb_position_px = thumb_pos / max * length_px;
+                    mouse_px.x - thumb_position_px
+                });
+            x           <- all4(&mouse_position,&drag_offset,&inner_length,&frp.set_max);
+            x           <- x.gate(&base_frp.is_dragging_track);
+            frp.jump_to <+ x.map(|(mouse_px,offset_px,length_px,max)| {
+                    let target_px = mouse_px.x - offset_px;
+                    target_px / length_px * max
+                });
+        }
+
+
+        // === Init Network ===
+
+        frp.set_length(200.0);
+        frp.set_thumb_size(0.2);
+        frp.set_max(1.0);
+        init_mouse_position.emit(Vector2(f32::NAN,f32::NAN));
+        init_color.emit(());
+    }
+}
+
+
+
+// ===========================
+// === Scrollbar Component ===
+// ===========================
+
+/// Scrollbar component that can be used to implement scrollable components. See module description
+/// for details.
```

Comment:
Again, I would put here an extended description, and refer to it from the module docs.